### DISCUSSION TITLE
Not displaying traffic with NetPrefix filter

### DISCUSSION THREAD

**Discussion by mellfice** [akvorado/akvorado#603] [Q&A] [answered]
Hello

I would like to express my thanks for such a great product.

I apologize for my English. 
 
After upgrading from version 1.7.2 to version 1.8.1 it does not show traffic when using NetPrefix. 

When a less specific network is specified in `NetPrefix` (/24), nothing is displayed (**image 1**). 
If you specify a more specific network in `NetPrefix` (/31), the traffic is shown (**image 2**). 

I also noticed if I use `SrcAddr` with `<<` operator, it shows the traffic (**image 3**). 
 
I tried to find it in documentation, but did not find anything. 
Maybe i dont understand correctly how to use filter?
In any case thanks!

image 1: ![image](https://user-images.githubusercontent.com/73892787/228301278-fb71eb8e-0424-4219-bddc-4298dd11c510.png)

image 2: ![image](https://user-images.githubusercontent.com/73892787/228301522-42669554-a059-42b0-8d3a-945ef09709df.png)

image 3: ![image](https://user-images.githubusercontent.com/73892787/228301710-daea9a3d-9dc3-4529-bd0a-8a27b6c1bcf4.png)

**Comment by vincentbernat** [comment, marked as answer]
When used as a filter, SrcNetPrefix needs an exact match. It looks like that's what you get. If you don't want an exact match, you have to use `SrcAddr <<` like you did.

**Reply by mellfice** [reply]
> When used as a filter, SrcNetPrefix needs an exact match. It looks like that's what you get. If you don't want an exact match, you have to use `SrcAddr <<` like you did.

Thank you for your response.

Is there an explanation of Filter `NetPrefix` somewhere in the documentation or elsewhere? 
I guess I don't understand it correctly

**Reply by vincentbernat** [reply]
There is not. Consider `SrcNetPrefix` as a string.
